contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+            - Phi (inplane rotation of sample, rotation axis is the samples 
+                  surface normal)
+
+
+          A sample normal centered description is as well possible(similar to XX):
+            - angle of incidence (angle between incident beam and sample surface)
+            - angle of detection (angle between detection beam and sample surface)
+            - angle of incident and detection beam (angle between the incident and scattering beam)
+            - angle of in-plane sample rotation (direction along the sample's surface normal)
+       
+        # For the sample centered approach, it has to be avoided, that the detection beam,
+        # incident beam or surface normal beam are parallel to each other
+        # (Is this a problem for Raman backscattering?)
+        # For beam centered approach (as done for beamlines), the grativation and beam direction are used to define
+        # the coordinate system. As beamlines are usually built flat - there is no case
+        # in which these two vectors can be parallel to each other
+        # This is different with the surface normal. Aside of the surface normal,
+        # at least one further additional natural direction is required.
+        # This could be the magnetic/geographic north pole
+        # magnetic north pole: easy to determine, but time depenent (long term)
+        # geographic north pole: harder to determine? (What about GPS?), but constant with time
+        
+        enumeration: [beam centered, sample normal centered]
+      wavelength_resolution(NXresolution):
         exists: optional
         doc: |
-          ISO8601 date when the instrument was constructed.
-          UTC offset should be specified.
-      software(NXprocess):
-        program:
+          The overall resolution of the optical instrument.
+        physical_quantity:
+          enumeration: [wavelength]
+        type:
+          exists: recommended
+        resolution(NX_FLOAT):
+          unit: NX_WAVELENGTH
           doc: |
-            Commercial or otherwise defined given name of the program that was
-            used to measure the data, i.e. the software used to start and
-            record the measured data and/or metadata.
-            If home written, one can provide the actual steps in the NOTE
-            subfield here.
-        version:
-          doc: |
-            Either version with build number, commit hash, or description of a
-            (online) repository where the source code of the program and build
-            instructions can be found so that the program can be configured in
-            such a way that result files can be created ideally in a
-            deterministic manner.
-        \@url:
-          exists: optional
-          doc: |
-            Website of the software.
-      firmware(NXprogram):
+            Minimum distinguishable wavelength separation of peaks in spectra.
+      instrument_calibration_DEVICE(NXsubentry):
```

Comment:
In MPES, we have started collecting calibrations within NXprocess (not as part of the instrument). Maybe something to consider.

Reply:
Why not use NXcalibrations here? Have a look at the[ `NXcalibration` base class](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXcalibration.html#nxcalibration). Some of these fields may exist in similar form there already. Alternatively, consider expanding `NXcalibration` so that the generic fields you define here can be used in other techniques as well.

Reply:
regarding NXprocess: I wanted to state here the calubration of the device/instrument itself - This was something from the first NXellipsometry/NXopt version. I have as well something to make dataset calibrations via NXprocess.

Ill as well have a look into NXcalibrations again.

Reply:
>  I wanted to state here the calubration of the device/instrument itself

As a suggestion: you can still keep it within NXinstrument, but use NXcalibration.